xarray/core/dataarray.py

```diff
@@ -2809,7 +2819,7 @@ def unstack(
         dim: Dims = None,
         fill_value: Any = dtypes.NA,
         sparse: bool = False,
-    ) -> DataArray:
+    ) -> Self:
```

Comment:
The comment above this method is now obsolete. (There should be more of these cases, just look for the mypy issue in the code)

Reply:
Great, lots of hits for `12846`!